admin/entities.py

```diff
@@ -103,6 +104,49 @@ class SubmissionAdminView(BaseAdminView):
 
 class OrganisationAdminView(BaseAdminView):
     _model = Organisation
+    instance = None
```

Comment:
I am not sure that storing the instance on a class variable is very safe in a web app. When we start handling multiple requests in parallel, the instances could get mixed up between requests.

The safer way of doing this in Flask is to use the `g` object, so maybe we can store this on `g.__organisation_admin_instance` or something - if we really need to have some reference that we can't track and pass between functions easily?

Reply:
Removed the instance from the class and added some tests.


A note regarding manually changing the form data in the html, as you've told me to do in our huddle last time:

1. So, any extra fields added to the POST request's form, which are in the `form_excluded_columns` list will be removed from the form by Flask-Admin. 
I tried it and the `test_organisationadmin_edit_disallowed` function tests this: it updates the allowed column, but removes all other columns from the form and won't update those. In this case we wouldn't even need the `on_model_change` function (but being there doesn't hurt anything).

2. To make absolutely sure that nothing can be changed with a forged post request, the `on_model_change` function aborts editing of an instance if there is any field in the form which is not in the `fields_allowed_to_edit` list. For this to be triggered the given columns should also be removed from the `form_excluded_columns` list (I don't know how it could be achieved though). This is tested in the `test_organisationadmin_on_model_change` function.

I hope it makes sense. =)